File: ea/src/evolution_engine/EvolutionLogger.java
package evolution_engine;

import evolution_engine.evolution.Individual;
import evolution_engine.fitness.FitnessResult;

import java.util.List;

/**
 * Created by itayaza on 27/10/2014.
 */
public interface EvolutionLogger {
    public void addGeneration(List<? extends Individual> individuals, FitnessResult fitness);
}

File: src/main/java/evolution_engine/EvolutionLogger.scala
package evolution_engine

import evolution_engine.evolution.Individual
import evolution_engine.fitness.FitnessResult

/**
 * Created by itayaza on 27/10/2014.
 */
trait EvolutionLogger[I <: Individual] {
  def addGeneration(individuals: List[I], fitness: FitnessResult[I])
}